docs/src/formula.md

```diff
@@ -14,167 +16,361 @@ fields with possibly heterogeneous types.  One of the primary goals of
+FormulaTerm
+Response:
+  y(unknown)
+Predictors:
+  1
+  a(unknown)
+  b(unknown)
+  c(unknown)
+  b(unknown) & c(unknown)
+
+julia> df = DataFrame(y = rand(9), a = 1:9, b = rand(9), c = repeat(["d","e","f"], 3))
+9×4 DataFrame
+│ Row │ y          │ a     │ b         │ c      │
+│     │ Float64    │ Int64 │ Float64   │ String │
+├─────┼────────────┼───────┼───────────┼────────┤
+│ 1   │ 0.236033   │ 1     │ 0.986666  │ d      │
+│ 2   │ 0.346517   │ 2     │ 0.555751  │ e      │
+│ 3   │ 0.312707   │ 3     │ 0.437108  │ f      │
+│ 4   │ 0.00790928 │ 4     │ 0.424718  │ d      │
+│ 5   │ 0.488613   │ 5     │ 0.773223  │ e      │
+│ 6   │ 0.210968   │ 6     │ 0.28119   │ f      │
+│ 7   │ 0.951916   │ 7     │ 0.209472  │ d      │
+│ 8   │ 0.999905   │ 8     │ 0.251379  │ e      │
+│ 9   │ 0.251662   │ 9     │ 0.0203749 │ f      │
+
+julia> f = apply_schema(f, schema(f, df))
+FormulaTerm
+Response:
+  y(continuous)
+Predictors:
+  1
+  a(continuous)
+  b(continuous)
+  c(DummyCoding:3→2)
+  b(continuous) & c(DummyCoding:3→2)
+
+julia> resp, pred = modelcols(f, df);
+
+julia> pred
+9×7 Array{Float64,2}:
+ 1.0  1.0  0.986666   0.0  0.0  0.0       0.0
+ 1.0  2.0  0.555751   1.0  0.0  0.555751  0.0
+ 1.0  3.0  0.437108   0.0  1.0  0.0       0.437108
+ 1.0  4.0  0.424718   0.0  0.0  0.0       0.0
+ 1.0  5.0  0.773223   1.0  0.0  0.773223  0.0
+ 1.0  6.0  0.28119    0.0  1.0  0.0       0.28119
+ 1.0  7.0  0.209472   0.0  0.0  0.0       0.0
+ 1.0  8.0  0.251379   1.0  0.0  0.251379  0.0
+ 1.0  9.0  0.0203749  0.0  1.0  0.0       0.0203749
+
+julia> coefnames(f)
+("y", ["(Intercept)", "a", "b", "c: e", "c: f", "b & c: e", "b & c: f"])
 
-## The `Formula` type
-
-The basic conceptual tool for this is the `Formula`, which has a left side and a
-right side, separated by `~`. Formulas are constructed using the `@formula` macro:
-
-```jldoctest
-julia> @formula(y ~ 1 + a)
-Formula: y ~ 1 + a
 ```
 
-Note that the `@formula` macro **must** be called with parentheses to ensure that
-the formula is parsed properly.
+Let's break down the formula expression ` y ~ 1 + a + b + c + b&c`:
+
+At the top level is the **formula separator** `~`, which separates the left-hand
+(or response) variable `y` from the right-hand size (or predictor) variables on
+the right `1 + a + b + c + b&c`.
+
+The left-hand side has one term `y` which means that the resopnse variable is
+the column from the data named `:y`.  The response can be accessed with the
+analogous `response(f, df)` function.
+
+The right hand side is made up of a number of different **terms**, separated by
+`+`: `1 + a + b + c + b&c`.  Each term corresponds to one or more columns in the
+generated model matrix: 
+
+* The first term `1` generates a constant or "intercept" column full of `1.0`s.
+* The next two terms `a` and `b` correspond to columns from the data table
+  called `:a`, `:b`, which both hold numeric data (`Float64` and `Int`
+  respectively).  Such **continuous terms** are converted to `Float64` and
```

Comment:
**numerical**?

Reply:
instead of "continuous terms"? or instead of numeric?

Reply:
Instead of "continuous"

Reply:
A numerical term isn't necessarily continuous.  For instance, you might want to treat a vector Ints as indices instead of numbers.

Reply:
But this distinction could be made clearer here I think...